src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/AuthController.java

```diff
@@ -42,16 +50,16 @@ public ResponseEntity getAuth(@RequestHeader("X-Client-Token") String clientToke
     @PostMapping("/auth")
     public AuthResponse postAuth(@RequestBody AuthModel authModel, @RequestHeader("X-Client-Token") String clientToken, HttpServletRequest request) throws IOException {
         UserModel userModel;
-        if (!request.getHeader("X-Client-Token").equals("")) {
+        if (userModelRepository.findByUserEmail(authModel.getEmail()) != null) {
+             userModel= userModelRepository.findByUserEmail(authModel.getEmail());
+        } else if (!request.getHeader("X-Client-Token").equals("")) {
```

Comment:
@adel-n `clientToken` check should be the first, and if there's none, we should find any authentication for the email and return the user for that